.github/workflows/delete-review-app.yml

```diff
@@ -0,0 +1,92 @@
+name: Delete Review App
+
+on:
+  pull_request:
+    types: [closed]
+    branches: [main]
+
+jobs:
+  delete-review-app:
+    name: Delete Review App ${{ github.event.pull_request.number }}
+    if: github.actor != 'dependabot[bot]'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Wait for Deploy App Workflow for review
+        id: wait_for_deployment
+        if: contains(github.event.pull_request.labels.*.name, 'deploy')
+        uses: fountainhead/action-wait-for-check@v1.0.0
+        with:
+         token: ${{ secrets.GITHUB_TOKEN }}
+         checkName: ${{ github.event.pull_request.number }} Deployment
+         ref: ${{ github.event.pull_request.head.sha }}
+         timeoutSeconds: 1800
+
+      - name: Exit whole workflow if wait was not successful
+        if: ${{ steps.wait_for_deployment.outputs.conclusion != '' && steps.wait_for_deployment.outputs.conclusion != 'success' }}
+        run: exit 1
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Terraform v0.13.5
+        uses: hashicorp/setup-terraform@v1.2.1
+        with:
+          terraform_version: 0.13.5
+
+      - name: Set Environment variables
+        run: |
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          TF_STATE_FILE=pr-$PR_NUMBER.tfstate
+          echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
+          echo "TF_STATE_FILE=$TF_STATE_FILE" >> $GITHUB_ENV
+
+          pr_state_file=$(az storage blob list -c review-paas-tfstate \
+           --connection-string $AZURE_STORAGE_CONNECTION_STRING \
+           --prefix $TF_STATE_FILE --query "[].name" -o tsv)
+
+          if [ ! -z "$pr_state_file" ]; then echo "TF_STATE_EXISTS=true" >> $GITHUB_ENV; fi;
```

Comment:
Cosmetic: we should be able to replace `if [ ! -z` with `if [ -n`

Reply:
Replaced